### PR TITLE
add audit SDK

### PR DESCRIPTION
**Pull request by jslching**

### CODE REVIEW

**Review by elv-serban — COMMENTED** [review]
Looks good to me.

File: src/ContentObjectAudit.js [hunk 1]
```diff
@@ -25,11 +25,11 @@ const ContentObjectAudit = {
     let audits = [];
     for(response of responses) {
       let url = new URL(response.url);
-      let res = await client.utils.ResponseToJson(response);
       let audit = { host: url.hostname };
       if(!response.ok) {
-        audit.error = res;
+        audit.error = JSON.stringify(response);
```
Comment (elv-nate) [inline src/ContentObjectAudit.js:30]: nit: I don't think there's a real reason to stringify the error here, it's fine just putting it in the error field of the audit.

File: src/HttpClient.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class HttpClient {
     allowRetry=true,
     uriIndex
   }) {
-    uriIndex = uriIndex || this.uriIndex;
+    if(uriIndex === undefined) { uriIndex = this.uriIndex; }
```
Comment (elv-nate) [inline src/HttpClient.js:66]: This line is unnecessary, it's cleaner to just deal with it in `BaseURI` (like you already do).

File: src/HttpClient.js [hunk 3]
```diff
@@ -191,18 +191,35 @@ class HttpClient {
...
-        method,
-        path,
-        queryParams,
-        body,
-        bodyType,
-        headers,
-        allowFailover: false,
-        uriIndex: this.uriIndex + i
-      }));
+      promises.push((async () => {
```
Comment (elv-nate) [inline src/HttpClient.js:207]: 👍

**Review by elv-nate — APPROVED** [review]
Looks good from my perspective. Minor nits, but that's it. That said, my review isn't very authoritative in this part of the code, so wait for Kevin's take.

File: src/ContentObjectAudit.js (hunk 1 above)
Comment (jslching) [reply src/ContentObjectAudit.js:30]: If I didn't do this, for some reason the error object doesn't get marshalled; it just returns "Object"

File: src/client/ContentAccess.js [hunk 4]
```diff
@@ -3086,13 +3086,11 @@ exports.VerifyContentObject = async function({libraryId, objectId, versionHash})
  * @param {string=} objectId - ID of the object
  * @param {string=} versionHash - Version hash of the object -- if not specified, latest version is returned
  * @param {string=} salt - base64-encoded byte sequence for salting the audit hash
- * @param {Array<number>=} samples - list of percentages used for sampling the content part list; up to 3 values max, 0.0-1.0 values
```
Comment (jslching) [inline src/client/ContentAccess.js]: Just FYI, the qfab API only allows up to 3 samples; if you specify more, it will fail

**Review by elv-nate — APPROVED** [review]
LGTM. Took me a minute to figure out what the filter was doing, but that is a clean one-liner to dedupe (for short arrays).

**Review by jslching — COMMENTED** [review]
LGTM, thanks!

File: src/ContentObjectVerification.js
Comment (jslching) [inline src/ContentObjectVerification.js:1]: Should we remove references to `ContentObjectVerification` as well?

File: src/client/ContentAccess.js (hunk 4 above)
Comment (elv-kevin) [reply src/client/ContentAccess.js]: got it, I will note it in the comment and silently truncate the list if it's too long

File: src/ContentObjectVerification.js
Comment (elv-kevin) [reply src/ContentObjectVerification.js:1]: Whoops, yeah I thought I did but there are still a bunch

**Pull request merged**